lib/ex_twilio/utilities/request_validator_plug.ex

```diff
@@ -0,0 +1,52 @@
+defmodule ExTwilio.Utilities.RequestValidatorPlug do
+  @moduledoc """
+  Plug to verify that requests are coming from Twilio using
+  `ExTwilio.Utilities.RequestValidator`
+
+  ## Example
+
+  Check that all requests are from Twilio, returning HTTP 401 if not.
+
+      defmodule MyRouter do
+        use Plug.Router
+
+        plug ExTwilio.Utilities.RequestValidatorPlug
+        plug :match
+        plug :dispatch
+
+        # Request has already been validated by the RequestValidatorPlug
+        post "/" do
+          # ... process call details ...
+          # ... build response using ExTwiml ...
+          conn
+          |> put_resp_header("content-type", "text/xml")
+          |> send_resp(200, response)
+        end
+      end
+  """
+
+  import Plug.Conn
+  alias ExTwilio.Utilities.RequestValidator
+
+  def init(opts), do: opts
+
+  def call(conn, _opts) do
+    case conn |> get_req_header("x-twilio-signature") |> List.first do
```

Comment:
This case is probably not needed since you get it for free with the validator check. I don t think there is enough improved benefit from doing this.